### ISSUE TITLE
Feature: add resize3d for NumPy volumes and CPU torch.Tensor volumes

### ISSUE THREAD

**Issue by ternaus** [albumentations-team/albucore#134]
## Problem

Volumetric transforms need one reusable, optimized primitive for resizing the spatial `(D, H, W)` axes. `Anisotropy3D` currently has to implement its own backend dispatch: NumPy uses sequential axis resizes by packing 1D signals into 2D images, while CPU Tensor volumes use true 3D trilinear interpolation. This belongs in Albucore rather than each 3D transform.

## Proposed API

```python
resize3d(
    volume: np.ndarray | torch.Tensor,
    size: tuple[int, int, int],
    interpolation: int = cv2.INTER_LINEAR,
    antialias: bool = False,
) -> np.ndarray | torch.Tensor
```

## Input/output contract

- NumPy: `(D, H, W)` and `(D, H, W, C)` volumes.
- CPU Tensor: channel-first `(C, D, H, W)` volumes.
- Resize only spatial axes; preserve channel count, dtype, device, and input representation.
- Support arbitrary channel counts, including `C > 4`.
- Return the requested spatial shape exactly.

## Implementation notes

- The NumPy path can use the existing Albucore 2D `resize` helper to resize one volume axis at a time. It should stay in Albucore so its packing strategy and C>4 handling are benchmarked once.
- The Tensor path should use native `torch.nn.functional.interpolate(..., mode="trilinear")`, without NumPy conversion.
- PyTorch does not currently expose `antialias=True` for 5D trilinear interpolation; downstream behavior should be documented or gated until upstream support lands: https://github.com/pytorch/pytorch/issues/191896

## Acceptance criteria

- Correctness matrix for uint8 and float32, channels 1/3/5, non-cubic volumes, and unit-length spatial axes.
- NumPy and Tensor public routes with representation-preserving output.
- Direct-kernel benchmarks for NumPy and Tensor routes, including C=5 and a comparison against a naive true-N-D implementation.
- A small set of transform integrations can then delegate their volumetric resize work to this primitive.

**Comment by ternaus** [comment]
Implementation status:\n\n- torch>=2.13.0 is now a required, locked dependency.\n- Public resize3d accepts NumPy DHWC and CPU torch.Tensor CDHW; Tensor execution is inference-only (no autograd).\n- The implementation includes full-path CPU benchmarks, contract/property tests, and public API documentation.\n\nUpstream capability records:\n- PyTorch 5D trilinear antialias: https://github.com/pytorch/pytorch/issues/191896\n- PyTorch CPU uint8 trilinear: https://github.com/pytorch/pytorch/issues/191907\n- OpenCV CPU volumetric DHWC resize: https://github.com/opencv/opencv/issues/29654\n- NumKong 3D DHWC resampling: https://github.com/ashvardanian/NumKong/issues/372\n\nDownstream extraction is intentionally sequenced after the next Albucore release: AlbumentationsX currently pins albucore==0.2.9, so replacing its local Anisotropy3D kernels before the release would break a normal downstream install.

**Comment by ternaus** [comment]
Done
